File: artisanat-backend/src/main/java/com/artisanat_backend/dto/SubOrderDto.java
package com.artisanat_backend.dto;

import com.artisanat_backend.model.Artisan;
import com.artisanat_backend.model.SubOrder;

import java.util.List;

/**
 * DTO for {@link SubOrder}
 */
public class SubOrderDto {
    double subTotal;
    Artisan artisan;
    OrderDto order;
    List<ProductDto> products;
}

File: artisanat-frontend/src/app/core/dtos/sub-order.dto.ts
import { ArtisanDTO } from './artisan.dto';
import { OrderDto } from './order.dto';
import { ProductDto } from './product.dto';

export interface SubOrderDto {
  subTotal: number;
  artisan: ArtisanDTO;
  order: OrderDto;
  products: ProductDto[];
}
